src/virtual-wan/azext_vwan/aaz/latest/network/vhub/_create.py

```diff
@@ -52,54 +52,84 @@ def _build_arguments_schema(cls, *args, **kwargs):
             options=["-n", "--name"],
             help="Name of the virtual hub.",
             required=True,
-            id_part="name",
-        )
-        _args_schema.location = AAZResourceLocationArg(
-            help="Location. Values from: `az account list-locations`. You can configure the default location using `az configure --defaults location=<location>`.",
-            required=True,
-            fmt=AAZResourceLocationArgFormat(
-                resource_group_arg="resource_group",
-            ),
         )
+
+        # define Arg Group "Properties"
+
+        _args_schema = cls._args_schema
         _args_schema.address_prefix = AAZStrArg(
             options=["--address-prefix"],
+            arg_group="Properties",
             help="CIDR address prefix for the virtual hub.",
         )
+        _args_schema.allow_b2b_traffic = AAZBoolArg(
+            options=["--allow-b2b-traffic"],
+            arg_group="Properties",
+            help="Flag to control branch-to-branch traffic for VirtualRouter hub.",
+        )
         _args_schema.hub_routing_preference = AAZStrArg(
             options=["--hub-routing-preference"],
+            arg_group="Properties",
             help="The hub routing preference gateway types.",
             enum={"ASPath": "ASPath", "ExpressRoute": "ExpressRoute", "VpnGateway": "VpnGateway"},
         )
         _args_schema.sku = AAZStrArg(
             options=["--sku"],
+            arg_group="Properties",
             help="The sku of the VirtualHub. Allowed values: Basic, Standard.",
         )
         _args_schema.asn = AAZIntArg(
             options=["--asn"],
+            arg_group="Properties",
             help="VirtualRouter ASN.",
             fmt=AAZIntArgFormat(
                 maximum=4294967295,
                 minimum=0,
             ),
         )
-        _args_schema.vwan = AAZResourceIdArg(
```

Comment:
@necusjz I'm using autogeneration with aaz-dev, and it's converting this argument from resource id to string. Is there a way I can fix this in the aaz model, or do I need to modify the generated output? To be clear the [rest api type](https://learn.microsoft.com/en-us/rest/api/virtualwan/virtual-hubs/create-or-update?tabs=HTTP) is a subresource so I'm not sure where string type is coming from.

Reply:
`.to_serialized_data()` could be helpful in this case.